src/api/app/controllers/webui/worker_capabilities_controller.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class Webui::WorkerCapabilitiesController < Webui::WebuiController
+  def show
+    cache_key = "worker_capabilities_#{params[:arch]}_#{params[:id]}"
+
+    xml_data = Rails.cache.fetch(cache_key, expires_in: 30.minutes) do
+      Backend::Api::BuildResults::Worker.capabilities(params[:arch], params[:id])
+    end || ''
```

Comment:
I don't think this would happen normally, to be honest it was mainly to cover a failing test. I'll keep things simple and remove this and the corresponding test.

Reply:
Never let tests leak into your implementation...